projects/plugins/crm/admin/settings/custom-fields.page.php

```diff
@@ -293,6 +293,89 @@
 	// Brutal update (note this is on top of updateActiveCustomFields DAL2+ work above)
 	$zbs->settings->update( 'customfields', $custom_fields );
 
+	/*
+	 * After the update, we must remove all references in the field sorting
+	 * settings for any custom fields that no longer exist. We begin with
+	 * a hard-coded map, which is required since field types in custom fields and
+	 * field sorts are hard-coded differently. Unifying them would demand major efforts.
+	 */
+	$sort_types_map = array(
+		'address'  => array(
+			'custom_key' => 'addresses',
+			'obj_key'    => 'zbsAddressFields',
+		),
+		'customer' => array(
+			'custom_key' => 'customers',
+			'obj_key'    => 'zbsCustomerFields',
+		),
+		'company'  => array(
+			'custom_key' => 'companies',
+			'obj_key'    => 'zbsCompanyFields',
+		),
+	);
+
+	/*
+	 * Creating a lookup table so we don't need to have multiple nested loops in
+	 * the next step. We do this by first adding all currently valid custom
+	 * fields, then adding all others (that should be the default ones).
+	 */
+	$custom_field_names = array();
```

Comment:
This variable name was throwing me off until I tested a bit deeper...really it returns an array of all fields for the given object, not just custom fields. Maybe `$sort_field_names` would be more readable?

Reply:
I agree. The comment above the variable name helps to some degree, though ideally I imagine we wouldn't be relying on the comment for an explanation of what the variable holds.

Reply:
Updated var name: 0f600c79c7